core/dispatch.c

```diff
@@ -752,9 +752,10 @@ dispatch_enter_dynamorio(dcontext_t *dcontext)
            wherewasi == DR_WHERE_APP ||
            /* If the thread was waiting at check_wait_at_safe_point when getting
             * suspended, we were in dispatch (ref i#3427). We will be here after the
-            * thread's context is being reset before sending it native.
+            * thread's context is being reset proactively (due to some -reset_at_*
+            * option) or before sending it native.
             */
-           (dcontext->go_native && wherewasi == DR_WHERE_DISPATCH));
+           wherewasi == DR_WHERE_DISPATCH);
```

Comment:
Could this instead keep the `go_native` and add to it "or the last exit was the special reset exit"?

Reply:
Done.